src/lib/route.js

```diff
@@ -94,11 +95,17 @@ class Route {
     const joinedCurve = this.curve.join(tailRoute.curve)
     const joinedHops = this.hops.concat(tailRoute.hops.slice(1))
 
+    // only put an expiry on the joined route if at least one of the head/tail
+    // routes expires too.
+    const joinedExpiresAt = (this.expiresAt || tailRoute.expiresAt) &&
```

Comment:
Wouldn't the expiry of the joined route be the minimum of these two, if both are defined?